src/main/java/org/venice/piazza/idam/util/GxOAuthClient.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright 2016, RadiantBlue Technologies, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.venice.piazza.idam.util;
+
+import model.security.authz.UserProfile;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.json.JacksonJsonParser;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.venice.piazza.idam.model.GxOAuthResponse;
+import util.PiazzaLogger;
+
+import javax.naming.InvalidNameException;
+import javax.naming.ldap.LdapName;
+import javax.naming.ldap.Rdn;
+import javax.servlet.http.HttpServletRequest;
+import java.util.Base64;
+import java.util.Map;
+
+@Component
+public class GxOAuthClient {
+
+	@Value("${GEOAXIS}")
+	private String gxBaseUrl;
+
+	@Value("${GEOAXIS_AUTH}")
+	private String gxAuthUrl;
+
+	@Value("${GEOAXIS_LOGOUT}")
+	private String gxLogoutUrl;
+
+	@Value("${GEOAXIS_CLIENT_ID}")
+	private String gxClientId;
+
+	@Value("${GEOAXIS_SECRET}")
+	private String gxClientSec;
+
+	@Autowired
+	private RestTemplate restTemplate;
+
+	@Autowired
+	private PiazzaLogger logger;
+
+	private static final String AUTHORIZATION = "Authorization";
+	private final String profileRequestUrl = "https://" + gxBaseUrl + "/ms_oauth/resources/userprofile/me";
+	private final String tokenRequestUrl = "https://" + gxBaseUrl + "/ms_oauth/oauth2/endpoints/oauthservice/tokens?grant_type=authorization_code&redirect_uri=";
```

Comment:
@russellorf Ah ok, so maybe we just put the full base url in vcap or separate environment variables for all of the following:
/userprofile/me
/tokens
/authorize
/logout

Reply:
yes